### PR TITLE
merging DependencyCollector and DependencyCollector.Shared

### PR DESCRIPTION
**Pull request by eddynaka**
## Changes
Merging DependencyCollector and DependencyCollector.Shared

### Checklist
- [X] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

**Comment by TimothyMothra** [comment]
I think the references to the Shared project will need to be removed from the Microsoft.ApplicationInsights.Web solution as well

File: WEB/Src/DependencyCollector/DependencyCollector/DependencyCollector.csproj [hunk 1]
```diff
@@ -41,6 +41,10 @@
     </PackageReference>
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Remove="Implementation\SqlClientDiagnosticSourceListener.cs" />
+  </ItemGroup>
```
Comment (TimothyMothra) [inline WEB/Src/DependencyCollector/DependencyCollector/DependencyCollector.csproj:46]: is this necessary?

Comment (eddynaka) [reply WEB/Src/DependencyCollector/DependencyCollector/DependencyCollector.csproj:46]: no, its not. just removed :)

**Comment by TimothyMothra** [comment]
@eddynaka there's one conflict with develop. please resolve and we can merge this today :)

**Pull request merged**